### PR TITLE
MAINTAINING.md: Several changes for clarity

### PR DESCRIPTION
**Pull request by danpetry**
### Contribution description

This PR includes several changes which clarify certain points, following feedback.

### CODE REVIEW

**Comment by miri64** [comment]
Clarified, that this PR is about the maintainer guidelines, not doc in general ;-)

File: MAINTAINING.md [hunk 1]
```diff
@@ -36,12 +36,14 @@ the overall validity of the PR.
     Is the problem that the PR intends to solve clearly stated?
 2.  Is the solution presented in the PR as simple as possible to satisfy the
     requirements, but no simpler?
-3.  Is the PR functionally atomic? Does it address a single issue or should
-    it be split into multiple individual PRs?
+3.  Is the PR a manageable size for you to handle? As a general rule of thumb,
```
Comment (kaspar030) [inline MAINTAINING.md]: While I agree that the requirement to be "atomic" might lead to single-line breakout PR's for three line changes, the rule of thumb should still be "as small as possible".  Where does that full day come from?

PR's are usually too large and rarely too small, thus exaggerating towards small might be beneficial. Also, it is much easier to cherry-pick more commits or join two PR's than to split them.

File: MAINTAINING.md [hunk 2]
```diff
@@ -36,12 +36,14 @@ the overall validity of the PR.
...
+    if the PR is likely to take more than a full day for one person to review,
+    then it may need to be broken down, at your discretion.
 4.  Is the solution well designed on a high level?
 5.  Do the concepts used by the PR make sense?
 6.  Does the PR break with existing concepts?
-7.  Is the structure of the PR itself valid?
+7.  Is there a clean commit history in the pulled branch? The commit history
```
Comment (kaspar030) [inline MAINTAINING.md]: Please rephrase the second sentence. The history is not about "the work that has been done", but about grouping the actual diff.

File: MAINTAINING.md (hunk 1 above)
Comment (danpetry) [reply MAINTAINING.md]: Ok thanks. Re the PR being runnable - would this mean that it doesn't, on its own, leave some area of the code in a broken state? For a new feature, we would say that the feature has to work after that one PR gets merged, and for a bugfix, the bug needs to be fixed in that PR and that PR only? And subject to these conditions, if the PR is able to be broken down and these still hold true, then it should be broken down?

Comment (bergzand) [reply MAINTAINING.md]: I tend to agree with @kaspar030 on this. Maybe something like "Is the PR confined to a single explainable task?"

> Keeping PRs manageable is more of an incentive for authors: excessive contributions have a hard time to get reviewed.

I think this is only one side of the argument. For maintainers it is easier to spot flaws and judge impact with contributions changing a single task.

File: MAINTAINING.md (hunk 2 above)
Comment (danpetry) [reply MAINTAINING.md]: What should the diff show/how should it be grouped?

**Comment by miri64** [comment]
Ping

File: MAINTAINING.md (hunk 2 above)
Comment (miri64) [reply MAINTAINING.md]: @kaspar030 @danpetry can you come to a conclusion here?

Comment (danpetry) [reply MAINTAINING.md]: I've pushed a change to this... I.e. I rephrased as asked. @miri64 could you check the below for correctness pls?

Comment (miri64) [reply MAINTAINING.md]: I'm okay with the current version, but since @kaspar030 requested the change I would give him a chance to react as well ;-).

Comment (danpetry) [reply MAINTAINING.md]: Ok, I think he will be ok with us proceeding, since no reply... would this then be ready to ACK?

**Comment by danpetry** [comment]
Great, thanks.

**Pull request merged**